src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -155,9 +155,13 @@ func (r *ClusterReconciler) Reconcile(
 	ar.headlessService()
 	ar.ingress()
 	ar.nodeportService()
+	pki, err := ar.getPKI()
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("creating pki: %w", err)
```

Comment:
NIT: from the API it is not easy to understand that `get` function (in this e.g. `getPKI`) is creating resource.

Reply:
yeah , the error message and the name of the function are a bit odd. So should it be getOrCreatePKI? or just createPKI?

Reply:
Yep, took a shortcut. Will update.